lib/SIL/BasicBlockUtils.cpp

```diff
@@ -101,7 +101,8 @@ void swift::getEdgeArgs(TermInst *T, unsigned edgeIdx, SILBasicBlock *newEdgeBB,
   case SILInstructionKind::BranchInst: {
     auto *B = cast<BranchInst>(T);
     for (auto V : B->getArgs())
-      args.push_back(V);
+      args.push_back(newEdgeBB->createPhiArgument(V->getType(),
+                                                  ValueOwnershipKind::Owned));
```

Comment:
What about cond_branch instructions?

I think the better fix would be to move createPhiArgument to the call site (into swift::splitEdge). Then all possible terminator instructions are handled by design.
This would also make the API cleaner: no need to pass newEdgeBB anymore.

Reply:
Actually, it would be sufficient to return the _number_ of arguments.

Reply:
Done